googleapis/src/test/java/io/grpc/googleapis/GoogleCloudToProdNameResolverTest.java

```diff
@@ -135,6 +139,10 @@ public HttpURLConnection createConnection(String url) throws IOException {
               new ByteArrayInputStream(("/" + ZONE).getBytes(StandardCharsets.UTF_8)));
           return con;
         } else if (url.equals(GoogleCloudToProdNameResolver.METADATA_URL_SUPPORT_IPV6)) {
+          if (supportIpV6) {
+            when(con.getInputStream()).thenReturn(
+                new ByteArrayInputStream("2600:2d00:ffb0:0".getBytes(StandardCharsets.UTF_8)));
```

Comment:
I don't think you are testing the case where there is an imput stream but it returns an empty response.

Reply:
Added a test case for empty stream.